packages/commons/src/interceptors/accessToken.ts

```diff
@@ -0,0 +1,30 @@
+import { AxiosInstance } from "axios";
+import ApiClientBase from "../core/ApiClientBase.js";
+
+export function configureAccessTokenInterceptor(
+  axios: AxiosInstance,
+  token: string,
+  headerName: string = "x-access-token",
+) {
+  axios.interceptors.request.use((conf) => {
+    if (!conf.headers.has(headerName)) {
+      conf.headers.set(headerName, token);
+    }
+    return conf;
+  });
+}
+
+export function withAccessToken<T extends ApiClientBase | AxiosInstance>(
+  client: T,
+  token: string | undefined,
```

Comment:
Why optional (and why in `configureAccessTokenInterceptor` not)?

Reply:
This makes it easier to use the function, i.e. in https://github.com/mittwald/api-client-js/pull/45/files#diff-e3af516cc4c91c89a315a806b8921d6cc79ab98f800f1954acf04fd41f5f619c

Reply:
`configureAccessTokenInterceptor` should not be exported